### ISSUE TITLE
Large grenade cassings dont split the reagents equaly between targets

### ISSUE THREAD

**Issue by Supermichael777** [tgstation/tgstation#28963]
To reproduce: put beaker of plasma in Large grenade casing with two grey cores.

Expected behavior: two slimes pop out

result: one slime and one core

**Comment by Fox-McCloud** [comment]
Also worth mentioning; extracts with multiple uses will only react once.

**Comment by XDTM** [comment]
Wait, you can put more than two beakers in large grenades?

**Comment by Fox-McCloud** [comment]
@XDTM 

No, it's debatably an exploit. You can put an unlimited amount of slime cores in a large grenade.

Just put the beakers in first (which limits you to two), then you can put in however many slime cores you want.

There's multiple ways this bug could be fixed; making it split the beaker contents over the slime cores (and slime core uses), or make it so the slime cores obey the same logic as beakers.

The problem with the latter is...well, what's the point when most of the dangerous reactions have a delay anyway and you can just inject+chuck; only real use is shoving a bunch of gold grenades in a box and setting them off all at once, really.

**Comment by octareenroon91** [comment]
From code analysis:
Slime extracts spawn with a capacity of 100u.
The grey slime plasma reaction destroys the slime extract immediately.
Large grenades transfer as much chem as possible into each extract in the grenade.
Large grenades transfer the contents of an extract after-reaction back into the grenade container _if the slime still exists._

A grenade with enough chems to deliver 100u including 1u plasma, and anything including another 1u plasma to a second slime extract, could produce two grey slimes.

**Comment by Lumipharon** [comment]
I had always presumed you could only fit one core into a large casing (making logical sense atleast), and having 50 gold slime reactions simultaneously from one nade is hilarious, if a bit BAD.

The advantage of slime nades over just hand reactions is the trigger mechanisms.
You can remotely activate them, voice activate, booby trap nades etc etc.

Nothing quite says 'don't steal my shit' like a bunch of angry hivebots/trees spawning out of your bag when someone opens it.

On a related note, can someone please bring back the teleport bluespace slime reaction? That shit was the greatest.

**Comment by Supermichael777** [comment]
@Lumipharon thats what mousetrap ttv bombs are for

**Comment by timothymtorres** [comment]
This is somewhat related to it #46464

**Comment by Sealed101** [comment]
the code is 10 years old in most of the juicy places pls fix

you can put an unlimited amount of slime cores into the nade because the `beakers` list is typed as `/obj/item/reagent_containers/cup`, which means a slime core never gets added to the list to be checked against the length of the list

https://github.com/tgstation/tgstation/blob/0f47384465c29911fef57a13013337f1ea6a2cb2/code/game/objects/items/grenades/chem_grenade.dm#L165-L185

since they never get added to that list, they won't show up on sci hud examine

you should not be able to stuff a beaker and more than one slime core in one of these, 
### HOWEVER 
the issue described is no longer reproducible, most likely due to #63393
